docs/API/metadata.md

```diff
@@ -1,43 +1,112 @@
+can set arbitrary values in the `user` namespace and those values are visible
+only to the same authenticated user. Other users may set different values for
+the same `user` namespace keys on the dataset, or completely different keys.
 
 ### Dataset namespace
 
-This defines the dataset resource, and contains metadata from the dataset tarball recieved from the Pbench agent, including the full contents of a `metadata.log` file which was generated when the agent packaged up the result data into a tarball.
+This defines the dataset resource, and contains metadata recieved from the
+Pbench Agent, including the full contents of a `metadata.log` file created
+while gathering results and during dataset packaging.
 
-This namespace includes the resource name, which can be modified by the owner of the dataset, but all other metadata in this namespace is controlled by the server.
+This namespace includes the resource name, which can be modified by the owner
+of the dataset. All other data in this namespace is controlled by the server
+and cannot be changed by the client.
 
 ### Server namespace
 
-This defines internal server state related to a dataset that's not inherent to the representation of the data. Most of this is not of much use to external clients, but can be observed.
+This defines internal Pbench Server management state related to a dataset
+that's not inherent to the representation of the user's performance metrics.
+Most of this is not of much use to external clients, but can be observed.
 
-Only `server.deletion` can be modified by an authenticated client on behalf of the owner of the dataset. When a dataset is uploaded, the server computes an expected time representing the user's default data retention period based on the time the dataset was uploaded. The owner of the dataset is allowed to modify this later, as long as the new timestamp is within the maximum allowed server data retention period.
+The exception is `server.deletion`, which is a time after which the Pbench
+Server may choose to delete the dataset. This is computed when a dataset is
+received based on user profile preferances and server configuration; but it can
+be modified by the owner of the dataset, as long as the new timestamp remains
+within the maximum allowed server data retention period.
 
 ### Dashboard namespace
 
-The server will never modify or directly interpret values in this namespace. An authenticated client representing the owner of a dataset can set any keys within this namespace to any desired values for retrieval later. All clients with read access to the dataset will see the same values.
+The server will never modify or directly interpret values in this namespace. An
+authenticated client representing the owner of a dataset can set any keys
+within this namespace to any valid JSON values (string, integer, boolean, list,
+or nested objects) for retrieval later. All clients with read access to the
```

Comment:
s/integer/number/  (I.e., we're not going to prohibit floating point numbers, are we?)